parcels/interaction/scipy_flat.py

```diff
@@ -0,0 +1,23 @@
+import numpy as np
+from scipy.spatial import KDTree
+
+from parcels.interaction.base_neighbor import BaseFlatNeighborSearch
+
+
+class ScipyFlatNeighborSearch(BaseFlatNeighborSearch):
+    name = "scipy"
+
+    def __init__(self, interaction_distance, interaction_depth):
+        super().__init__(interaction_distance, interaction_depth)
+
+    def find_neighbors_by_coor(self, coor):
+        coor = coor.reshape(3, 1)
+        corrected_coor = (coor/self.inter_dist).reshape(-1)
+        rel_idx = np.array(self._kdtree.query_ball_point(corrected_coor, r=1))
+        neighbor_idx = self.active_idx[rel_idx]
+        return neighbor_idx, np.vstack(self._distance(coor, neighbor_idx))
+
+    def rebuild(self, values=None, active_mask=-1):
+        super().rebuild(values, active_mask)
+        self._corrected_values = values[:, self._active_idx]/self.inter_dist
+        self._kdtree = KDTree(self._corrected_values.T)
```

Comment:
@erikvansebille : well, that is an expert trick from @qubixes here, I think. scikit-spatial's KDTree exists as a slow Python-based version for ANY array, and as high-speed C-based tree (called `cKDTree`) if the array to be tree'd is specifically C-contiguous. Therefore, this transpose will allow us to use the fast tree version.

Further infos:
https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.KDTree.html
https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.cKDTree.html

Reply:
@CKehl If it works that way, then that would be great.. If I remember correctly, I just had to transpose because that's the way Scipy expects dimensions vs points. It could be that less transposing is needed for the kdtree specifically, though I'm not sure how much that will actually save (also since other neighbor search types  might prefer [dim, n_p] ordering. My intuition says that the difference will not be so big, but I'm not sure.